Alerting hipchat

jregovic opened the pull request:
https://github.com/grafana/grafana/issues/6451

Adds an alert notification using the HipChat v2 room API.
Messages go to the v2/room/ROOMID/notification?auth_token=AUTHTOKEN endpoint

torkelo committed:
updated websocket lib

torkelo committed:
feat(live): just wanted to checkout how far I got on the websocket data source

torkelo committed:
feat(live): minor progress on stream manager

torkelo committed:
refactor(http): refactoring http server

xginn8 committed:
adding refresh url param to pass refresh interval #7124

ryu1-sakai committed:
Enable image rendering to use browser time offset

bergquist committed:
tech(build): adds package signing step

bergquist committed:
tech(build): updates path to signing file

bergquist committed:
tech(build): sign and deploy packages to s3

Commit:
Adding HipChat notifier

Fixed syntax error in RoomId prventing compile
Set author

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=7374) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=7374) before we can accept your contribution.<br/>**28** out of **29** committers have signed the CLA.<br/><br/>:white_check_mark: bergquist<br/>:white_check_mark: torkelo<br/>:white_check_mark: utkarshcmu<br/>:white_check_mark: DanCech<br/>:white_check_mark: mtanda<br/>:white_check_mark: vitaliyf<br/>:white_check_mark: simnv<br/>:white_check_mark: vaibhavinbayarea<br/>:white_check_mark: Ricky-N<br/>:white_check_mark: leonoff<br/>:white_check_mark: mekalz<br/>:white_check_mark: theisenmark<br/>:white_check_mark: jaxxstorm<br/>:white_check_mark: andrewhowdencom<br/>:white_check_mark: xginn8<br/>:white_check_mark: adrianlzt<br/>:white_check_mark: tomasv<br/>:white_check_mark: connection-reset<br/>:white_check_mark: huydx<br/>:white_check_mark: smalik03<br/>:white_check_mark: rrickardt<br/>:white_check_mark: oliverpool<br/>:white_check_mark: lpic10<br/>:white_check_mark: FFalcon<br/>:white_check_mark: chrisbenti<br/>:white_check_mark: ctide<br/>:white_check_mark: Tepa00<br/>:white_check_mark: ryantxu<br/>:x: James Regovich<hr/>**James Regovich** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

torkelo commented:
this needs rebase :)

solugebefola committed:
Disable sign out for AuthProxyEnabled

Sign out was disabled for AuthProxyEnabled in #3122.  This is to catch the one other instance of a sign out link in the side menu.

bergquist commented:
master have now been merged into grafana/alerting_hipchat.

Please rebase your changes from grafana/alerting_hipchat if you want to target alerting_hipchat.

bergquist committed:
Merge pull request #7377 from solugebefola/patch-3

Disable sign out for AuthProxyEnabled

bergquist committed:
docs(changelog): adds note about closing #7377

bergquist committed:
tech(build): create sha files after signing packages

bergquist committed:
style: go fmt with simplify

bergquist committed:
Merge branch 'new_deploy'

kitwestneat committed:
file_export.ts: #5325 missed exportTableDataToCsv when adding sep (#7391)

lpic10 committed:
Set time when testing ES datasource, fixes #7044 (#7390)

okushchenko committed:
add DeliveryStreamName dimension to AWS/Firehose Cloudwatch namespace (#7388)

torkelo commented:
what is the alerting_hipchat branch?

jregovic commented:
alerting_hipchat is the branch created for https://github.com/grafana/grafana/issues/6451. My changes add the hook to HipChat.

torkelo committed:
fix(): conf file had left over of old config key, #7395

bergquist commented:
@torkelo it was requested the make collaboration easier. 

This PR still need to rebase against the `alerting_hipchat` branch.

torkelo committed:
feat(packaging): Make config file not part of package to simplify upgrade, made commit always a short sha, changed RESTART_ON_UPGRADE to true by default, #7399, #7227

jaylinski committed:
Remove deprecated dataproxy logging setting (#7398)

This should have been removed in 06440ef57b3a7d868de93dae694ec542218db092, but was probably forgotten.

Solves #6718.

bergquist committed:
tech(build): deploy builds for each commit to master

bergquist committed:
tech(build): remove sha calc from build step

bergquist committed:
tech(build): remove unused deps

bergquist committed:
tech(build): fixes broken build

bergquist committed:
tech(build): option to not include buildIteration in package name

bergquist committed:
tech(build): dont include buildnumber when building tags

bergquist committed:
tech(build): disable build based on tag

bergquist committed:
tech(build): disable builds for master

bergquist committed:
tech(build): disable package iteration for release builds

jregovic commented:
I'm pretty sure I've rebased this correctly. Am I missing something?

Erliz committed:
#7383 Fix factory typo in telegram notifier (#7406)

dpavlos committed:
[3743] Adds support for user search based on login or email (#7334)

* [3743] Adds support for user search based on login or email

* Use query parameter instead

* Use macaron Query() instead

torkelo committed:
Merge branch 'time-offset-to-image' of https://github.com/ryu1-sakai/grafana into ryu1-sakai-time-offset-to-image

torkelo committed:
feat(server side png rendering): added timezone parameter for server side rendering, refactoring PR #7264

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

torkelo committed:
refactor(): refactoring PR #7141

torkelo committed:
fix(): fixed unit test broken in last PR merge

torkelo committed:
refactor(): refactoring PR #7124

torkelo committed:
Merge branch 'xginn8-master'

torkelo committed:
fix(): another unit test fix attempt

torkelo committed:
hiding grafana live ds

torkelo committed:
Merge branch 'master' into websocket

torkelo committed:
mergefix(): fixed issue with master merge

bergquist committed:
tech(build): prepare for tag builds

torkelo committed:
fix(packaging): fixed issue in deb package post install script that made it think upgrade was happening, fixes #7403

Erliz committed:
Add image url to telegram alert notifier

If public image url of alert is exist add it to the telegram message

kev009 committed:
Add an idiomatic plugin path for FreeBSD (#7410)

bergquist committed:
Merge pull request #7417 from Erliz/patch-1

Add image url to telegram alert notifier

mtanda committed:
support regex expansion in templating "Regex" field (#6565)

bergquist committed:
docs(changelog): adds note about closing #7417

bergquist committed:
tech(build): deploy to new s3 buckets

bergquist committed:
tech(build): upgrade golang to 1.7.4

bergquist committed:
tech(build): update windows trigger

bergquist committed:
tech(build): enable windows build trigger

bergquist committed:
tech(build): fixes invalid script file usage

bergquist committed:
tech(build): starts docker build for latest build

bergquist commented on pkg/services/alerting/notifiers/hipchat_test.go:
```diff
@@ -0,0 +1,81 @@
+				So(hipchatNotifier.Type, ShouldEqual, "hipchat")
+				So(hipchatNotifier.Url, ShouldEqual, "http://google.com")
+				So(hipchatNotifier.ApiKey, ShouldEqual, "")
+				So(hipchatNotifier.RommId, ShouldEqual, "")
```
Typo. This doesn't compile.

jregovic replied:
fixed.

bergquist reviewed (changes requested):
The commit author have to be the same as the github user. Otherwise you cannot sign the CLA which is required before we merge any PR

bergquist committed:
tech(build): slimdown the size of the build container

bergquist committed:
feat(alerting): changes default timerange to 15min

bergquist committed:
tech(build): use build number as iteration number

torkelo committed:
fix(graph legend): fixed legend table mode scrollbar visible when it needs to to, fixes #6828

bergquist committed:
tech(build): fixes broken build

lucapette committed:
Fix typo

bergquist committed:
Merge pull request #7442 from lucapette/patch-1

Fix typo

bergquist committed:
docs: update nodejs version requirement in readme

bergquist committed:
chore(vet): fixes invalid fmt.Sprintf format

bergquist committed:
tech(build): require go vet to pass

jifwin committed:
Fix requests cancelling (#7457)

* fix backendSrv request cancelling

* revert imports

* formatting

torkelo committed:
Update building docs readme

bergquist committed:
tech(alerting): adds tags to alerting data model

bergquist committed:
feat(alerting): transform labels into tags for prometheus tsdb

bergquist committed:
tech(build): echo before running go fmt/vet

torkelo committed:
docs(): fixing redirects for moved doc pages, updating links to point to new urls

Commit:
Fixed missed error in test

Commit:
Fix to formatting

bergquist commented:
This is starting to get out of hand :) I suggest you move the code for the alerting notifier to a new branch and create a new pull request that targets the master branch. 

The hipchat branch was created to make collaboeration with others easier but since its only you working on it its much easier to just target the master branch.

Closing this since 97 files changes is way to much to review. But please open a new pull request!
